travelling_salesman.py

```diff
@@ -102,12 +125,29 @@ def add_moves(self) -> Iterable[Component]:
         for city in self.unvisited_cities:
             yield Component(self.visited_cities[-1], city)
 
+    # https://tsp-basics.blogspot.com/2017/03/3-opt-iterative-general-idea.html
```

Comment:
I believe leaving the link in the issue/PR here is enough

Reply:
Removed